### ISSUE TITLE
Build fauilure with certain features

### ISSUE THREAD

**Issue by jfrimmel** [rhaiscript/rhai#383]
Hello, I've found a build error while restricting the language features available for the users of my application. Reproduction is written below. I think, this is a bug.

```bash
cargo new test2
cd test2
echo 'rhai = { version = "0.19", features=["no_index", "no_object", "no_function", "no_module", "no_closure"] }' >> Cargo.toml
cargo check
```
Output:
```
    Checking rhai v0.19.14
error[E0599]: no method named `set_value` found for enum `Target<'_>` in the current scope
    --> /home/jfrimmel/.cargo/registry/src/github.com-1ecc6299db9ec823/rhai-0.19.14/src/engine.rs:1936:20
     |
287  | pub enum Target<'a> {
     | ------------------- method `set_value` not found for this
...
1936 |             target.set_value(new_value, new_value_pos)?;
     |                    ^^^^^^^^^ help: there is an associated function with a similar name: `is_value`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `rhai`

To learn more, run the command again with --verbose.
```

**Comment by schungx** [comment]
Ah.  Thanks for catching this!

**Comment by schungx** [comment]
This is now fixed in the latest master.  Please pull from here to check.

**Comment by jfrimmel** [comment]
Confirmed fixed using `fb633f67` or crate version `0.19.15`.
